app/src/main/java/com/sun_asterisk/moviedb_50/screen/listmovie/ListMovieFragment.kt

```diff
@@ -19,27 +41,145 @@ class ListMovieFragment : Fragment(), ListMovieContract.View {
+                listMovieProgressBarLayout.visibility = View.VISIBLE
+            } else {
+                listMovieProgressBarLayout.visibility = View.GONE
+                loadingProgressBar.visibility = View.GONE
+            }
+        }
     }
 
     private fun initPresenter() {
-        presenter?.setView(this)
+        presenter.setView(this)
         arguments?.run {
             val type = getString(Constant.BASE_TYPE)
             val value = getString(Constant.BASE_VALUE)
-            if (type != null && value != null) presenter?.getMovies(type, value)
+            if (type != null && value != null) presenter.getMovies(
+                type,
+                value,
+                Constant.BASE_PAGE_DEFAULT
+            )
+        }
+    }
+
+    private fun initAdapter() {
+        view?.run {
+            listMovieRecyclerView.setHasFixedSize(true)
+            listMovieRecyclerView.adapter = adapter.apply {
+                onItemClick = { item, _ ->
+                    addFragment(item)
+                }
+            }
+            listMovieRecyclerView.addOnScrollListener(object : OnScrollListener() {
+                override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+                    super.onScrollStateChanged(recyclerView, newState)
+                    if (newState == AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL) {
+                        isScrolling = true
+                    }
+                }
+
+                override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+                    super.onScrolled(recyclerView, dx, dy)
+                    val linearLayoutManager: LinearLayoutManager =
+                        listMovieRecyclerView.layoutManager as LinearLayoutManager
+                    if (isScrolling && linearLayoutManager.findLastVisibleItemPosition() ==
+                        adapter.itemCount - 1
+                    ) {
+                        isScrolling = false
+                        if (currentPage + 1 <= totalPage) {
+                            loadingProgressBar.visibility = View.VISIBLE
+                            currentPage += 1
+                            arguments?.run {
+                                val type = getString(Constant.BASE_TYPE)
+                                val value = getString(Constant.BASE_VALUE)
+                                if (type != null && value != null) presenter.getMovies(
+                                    type,
+                                    value,
+                                    currentPage
+                                )
+                            }
+                        }
+                    }
+                }
+            })
+        }
+    }
+
+    private fun initToolBar() {
+        view?.let {
+            val toolbar = it.findViewById<Toolbar>(R.id.toolbar)
+            (activity as MainActivity).run {
+                setSupportActionBar(toolbar)
```

Comment:
```suggestion
                setSupportActionBar(toolbar_base.toolbar)
```
bỏ cái findViewById đi.